app/kubernetes-ingress-controller/2.0.x/guides/using-udpingress.md

```diff
@@ -0,0 +1,323 @@
+Once you've made the necessary configurations you can apply your changes:
+
+```
+$ helm upgrade --namespace <NAMESPACE> --version <CHART_VERSION> -f values.yaml <RELEASE_NAME> kong/kong
+```
+
+Replace `<NAMESPACE>`, `<CHART_VERSION>` and `<RELEASE_NAME>` with the values you deployed Kong with.
+
+Alternatively if you are using only command line flags to deploy and manage Kong
+the same configuration can be achieved with flags:
+
+```shell
+$ helm upgrade --namespace <NAMESPACE> --version <CHART_VERSION> <RELEASE_NAME> kong/kong \
+  --set "udpProxy.enabled=true" \
+  --set "udpProxy.type=LoadBalancer" \
+  --set "udpProxy.stream[0].containerPort=9999"
+  --set "udpProxy.stream[0].servicePort=9999"
+  --set "udpProxy.stream[0].protocol=UDP"
+```
+
+Wait for the `LoadBalancer` service to be ready for the Gateway (see `kubectl get services`)
+and once it's up Kong is ready to serve UDP traffic on port `9999` external to the cluster
+using `UDPIngress` resources.
+
+## Deploying UDPIngress
+
+Now that Kong is listening on `9999` we can create a `UDPIngress` resource which will attach
+our CoreDNS service to that port so we can make DNS requests to it from outside the cluster.
+
+```yaml
+apiVersion: configuration.konghq.com/v1beta1
+kind: UDPIngress
+metadata:
+  name: minudp
+  namespace: udpingress-example
+  annotations:
+    kubernetes.io/ingress.class: kong
+spec:
+  rules:
+  - backend:
+      serviceName: coredns
+      servicePort: 53
+    port: 9999
+```
+
+The above binds the Kong Gateway port `9999` to the `Service` port `53` for our DNS server.
+
+Save the file as `coredns-udpingress.yaml` and apply it:
+
+```shell
+$ kubectl apply -f coredns-udpingress.yaml
+```
+
+We can now make DNS requests via Kong.
+
+## Verification
+
+The exercise is complete and all that's left to do is verify that everything is working
+by making a DNS request to our CoreDNS server.
+
+This example assumes you have the `dig` command available on your local system, but if
+you don't look to your operating system documentation for a similar DNS lookup tool.
+
+Firstly you'll need to retrieve the IP address of the UDP load balancer service that we
+configured in previous sections:
+
+```shell
+$ export KONG_UDP_ENDPOINT="$(kubectl -n <NAMESPACE> get service <RELEASE_NAME>-kong-udp-proxy \
+    -o=go-template='{{range .status.loadBalancer.ingress}}{{.ip}}{{end}}')"
+```
+
+Note that in the above you'll need to replace `<NAMESPACE>` and `<RELEASE_NAME>` with the values
+you originally provided to `helm install` in your own environment.
+
+Now that we've stored the IP in the environment variable `KONG_UDP_ENDPOINT` we can use
+that with `dig` to make a DNS lookup through the CoreDNS server we set up and exposed
+using `UDPIngress`:
+
+```shell
+$ dig @${KONG_UDP_ENDPOINT} konghq.com
+;; ANSWER SECTION:
+konghq.com.		30	IN	A	34.83.126.248
+
+;; Query time: 60 msec
+;; SERVER: <KONG_UDP_ENDPOINT>#9999
+;; WHEN: Thu Aug 19 08:39:16 EDT 2021
+;; MSG SIZE  rcvd: 77
+```
```

Comment:
Split the command from the response:

{% navtabs codeblock %}
{% navtab Command %}
```shell
dig @${KONG_UDP_ENDPOINT} konghq.com
```
{% endnavtab %}
{% navtab Response %}
```shell
;; ANSWER SECTION:
konghq.com.		30	IN	A	34.83.126.248
;; Query time: 60 msec
;; SERVER: <KONG_UDP_ENDPOINT>#9999
;; WHEN: Thu Aug 19 08:39:16 EDT 2021
;; MSG SIZE  rcvd: 77
```
{% endnavtab %}
{% endnavtabs %}

Reply:
Changed in 9d4f5af99201d9ff45b71705d220b595226c279f